### PR TITLE
Java lab hash

### PR DESCRIPTION
**Pull request by wafman**
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

### CODE REVIEW

File: code401Challenges/src/main/java/code401Challenges/hashtable/Hashtable.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            Node temp = map[hashKey];
+            map[hashKey] = node;
+            map[hashKey].setNext(temp);
+        }
+    }
+
+    public String get(String key){
+        int searchHashKey = hash(key);
+        while(this.map[searchHashKey] != null){
+            if(this.map[searchHashKey].getKey() == key){
```
Comment (evanslaton) [inline code401Challenges/src/main/java/code401Challenges/hashtable/Hashtable.java]: You should definitely use `.equals()` instead of `==`. Remember, `==` checks to see if the two things being compared a literally the same (the same thing stored in the same place in memory). `.equals()` checks to see if things are logically equal -the things being compared have the same value but aren't literally the same thing stored in memory.

This probably still passed your tests because Java has a built in behavior when using hard coded Strings where it uses the exact same string, stored in the same place in memory instead of creating 2 string and storing them separately.

File: code401Challenges/src/main/java/code401Challenges/hashtable/Hashtable.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                return this.map[searchHashKey].getValue();
+            }
+            this.map[searchHashKey] = this.map[searchHashKey].getNext();
```
Comment (evanslaton) [inline code401Challenges/src/main/java/code401Challenges/hashtable/Hashtable.java]: I think this line is actually writing over data causing you to lose whatever was previously stored at `this.map[searchHashKey]`
